xml-reader/src/main/java/org/triplea/generic/xml/reader/XmlParser.java

```diff
@@ -0,0 +1,109 @@
+package org.triplea.generic.xml.reader;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import javax.annotation.Nonnull;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.java.Log;
+import org.triplea.java.function.ThrowingRunnable;
+
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+@Log
+class XmlParser {
+  @Nonnull private final String tagName;
+
+  public static XmlParser tag(final String tagName) {
+    return new XmlParser(tagName.toUpperCase());
+  }
```

Comment:
I don't think this method is used anymore?

Reply:
No, artifact of this still being rough and a draft. Now that XmlParser is effectively an internal class, I simplified it a good bit. Having the fluent-API is no longer that useful since it's only used from the one class now.

The reasoning for the fluent-API is to avoid adding tag handles and body handlers together. If an XML has nested tags, then it should not have body content and vice versa. Adding both kinds of handlers is generally an error and hence the reason for an API that avoided that configuration.